draft-ietf-httpbis-connect-tcp.md

```diff
@@ -212,8 +273,32 @@ Template-driven TCP proxying is largely subject to the same security risks as cl
 
 A small additional risk is posed by the use of a URI Template parser on the client side.  The template input string could be crafted to exploit any vulnerabilities in the parser implementation.  Client implementers should apply their usual precautions for code that processes untrusted inputs.
 
+## Resource Exhaustion attacks
+
+Proxy implementors should take special care to avoid resource exhaustion attacks when the client is not trusted. A malicious client can achieve highly asymmetric resource usage by colluding with a destination server and violating the ordinary rules of TCP or HTTP.  Some example attacks and mitigations:
```

Comment:
Do you need the `is not trusted` caveat? Maybe even deleting the entire sentence leaves the paragraph clear enough that there are client-initiated resource exhaustion attacks a proxy needs to watch out for.

NB: I don't tend to trust anything at this layer of the stack, since it can be controlled by untrusted code running at a higher layer

Reply:
OK, I shortened this considerably.